src/universal_transfer_operator/datasets/dataframe/base.py

```diff
@@ -10,12 +12,17 @@ class Dataframe(Dataset):
     """
     Repersents all dataframe dataset.
     Intended to be used within library.
-
-    :param path: Path to a file in the filesystem/Object stores
-    :param conn_id: Airflow connection ID
     :param name: name of dataframe
+
     """
 
-    name: str | None = None
+    name: str = field(default="")
+    dataframe: Any = field(default=None)
+    uri: str = field(init=None)  # type: ignore
 
     # TODO: define the name and namespace for dataframe
+
+    @uri.default
+    def _path_to_dataset_uri(self) -> str:
```

Comment:
How do you make sure the path to the dataframe dataset uri is unique?

Reply:
I don't have a good option for this, hence kept it open. A few options that I considered are
1. Hash of that data, but that can change.
2. Memory address. But this needs to be reconstructed for different dag which might be running on different machines.

I'm open to suggestions here.

Reply:
I like approach one.

Reply:
@sunank200  I don't think we have a URI for something that only exists in memory because of the reason listed about. That's why users can create a unique string for themselves or we can generate that for them. But If we have a better solution we can implement that.